clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/transport/HttpsTransportClient.java

```diff
@@ -2,21 +2,54 @@
 
 import com.linkedin.venice.exceptions.VeniceException;
 import com.linkedin.venice.security.SSLFactory;
-import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
-import org.apache.http.impl.nio.client.HttpAsyncClients;
-import org.apache.http.nio.conn.ssl.SSLIOSessionStrategy;
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClients;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManagerBuilder;
+import org.apache.hc.client5.http.ssl.ClientTlsStrategyBuilder;
+import org.apache.hc.core5.http2.HttpVersionPolicy;
 
 
 public class HttpsTransportClient extends HttpTransportClient {
   private SSLFactory sslFactory;
 
-  public HttpsTransportClient(String routerUrl, SSLFactory sslFactory) {
-    this(
-        routerUrl,
-        HttpAsyncClients.custom().setSSLStrategy(new SSLIOSessionStrategy(sslFactory.getSSLContext())).build());
+  private boolean requireHTTP2;
+
+  public HttpsTransportClient(
+      String routerUrl,
+      int maxConnectionsTotal,
+      int maxConnectionsPerRoute,
+      boolean requireHTTP2,
+      SSLFactory sslFactory) {
+    this(routerUrl, buildClient(maxConnectionsTotal, maxConnectionsPerRoute, requireHTTP2, sslFactory));
+    this.maxConnectionsTotal = maxConnectionsTotal;
+    this.maxConnectionsPerRoute = maxConnectionsPerRoute;
+    this.requireHTTP2 = requireHTTP2;
     this.sslFactory = sslFactory;
   }
 
+  private static CloseableHttpAsyncClient buildClient(
+      int maxConnectionsTotal,
+      int maxConnectionsPerRoute,
+      boolean requireHHtp2,
+      SSLFactory sslFactory) {
+    if (requireHHtp2) {
+      // HTTP2 only client doesn't allow you to configure the ConnectionManager
+      return HttpAsyncClients.customHttp2()
```

Comment:
Can we leverage this utility class to instantiate a httpclient5 based h2 client?
https://github.com/linkedin/venice/blob/main/internal/venice-client-common/src/main/java/com/linkedin/venice/httpclient5/HttpClient5Utils.java

The utility class has been proved inside LinkedIn.